packages/bundler/src/BundlerServer.ts

```diff
@@ -30,20 +32,34 @@ export class BundlerServer {
     // eslint-disable-next-line @typescript-eslint/no-misused-promises
     this.app.post('/rpc', this.rpc.bind(this))
 
-    this.app.listen(this.config.port)
+    this.httpServer = this.app.listen(this.config.port)
+    this.startingPromise = this._preflightCheck()
   }
 
-  async preflightCheck (): Promise<void> {
+  startingPromise: Promise<void>
```

Comment:
I don't understand why does this server need to keep around this promise of itself starting. Can you please explain why?